### PR TITLE
Added B4X programming language

### PR DESCRIPTION
**Pull request by AnywhereSoftware**
Added B4X programmaing language

## Description
Add B4X programming language

## Checklist:

- [X] **I am adding a new language.**
  - [X] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      - https://github.com/search?q=extension%3Ab4j+sub&type=Code
      -  https://github.com/search?q=extension%3Ab4a+sub&type=Code
    b4r and b4i are more rare but they are unique to B4X.
  - [X] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - A cross platform (B4A, B4J and B4i) project is included in the samples folder.
    - Sample license(s): MIT. If needed I can make it public domain. I wrote the sample.
  
  - [ ] I have included a syntax highlighting grammar: https://github-lightshow.herokuapp.com/
  - [ ] I have updated the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -435,6 +435,17 @@ Awk:
...
   ace_mode: text
   language_id: 28
+B4X:
+  type: programming
+  color: "#5a6986"
+  extensions:
+  - ".b4a"
+  - ".b4j"
+  - ".b4i"
+  - ".b4r"
```
Comment (lildude) [inline lib/linguist/languages.yml:445]: Please remove the extensions that you've stated don't meet the usage requirements

**Review by lildude — CHANGES_REQUESTED** [review]
Along with removing the low-usage extensions as commented inline, please remove all sample files from the PR that do not have the extensions you're adding support for.

**Comment by lildude** [comment]
Looks like the current usage on GitHub.com for both extensions proposed doesn't meet the minimum requirements for inclusion yet:

- `b4j`
   Total files found: 561
   Unique public user/repos: 52
   Unique owners: 30

- `b4a`
   Total files found: 355
   Unique public user/repos: 124
   Unique owners: 89

**Review by pyhoon — APPROVED** [review]

**Review by pyhoon — APPROVED** [review]

**Comment by lildude** [comment]
Closing this as stale as it has been sitting without the requested changes being made for fast approaching two years. Usage is also nowhere near ready for inclusion.

Please open a new and complete PR when usage has met required levels.

**Pull request closed without merging**